rust/lsp-lib/src/language_server_client.rs

```diff
@@ -217,10 +274,59 @@ impl LanguageServerClient {
                 uri: self.opened_documents.get(&view_id).unwrap().clone(),
             },
         };
-
         let params = Params::from(serde_json::to_value(text_document_did_save_params).unwrap());
         self.send_notification("textDocument/didSave", params);
     }
+
+    pub fn request_hover<CB>(
+        &mut self,
+        view_id: ViewId,
+        position: Position,
+        on_result: CB,
+    ) where
+        CB: 'static + Send + FnOnce(&mut LanguageServerClient, Result<Value, Error>),
+    {
+        let text_document_position_params = TextDocumentPositionParams {
+            text_document: TextDocumentIdentifier {
+                uri: self.opened_documents.get(&view_id).unwrap().clone(),
+            },
+            position,
+        };
+
+        let params = Params::from(serde_json::to_value(text_document_position_params).unwrap());
+        self.send_request("textDocument/hover", params, Box::new(on_result))
+    }
+
+    pub fn request_definition<CB>(&mut self, view_id: ViewId, position: Position, on_result: CB)
```

Comment:
these two functions probably don't belong here?

Reply:
Removing this bit.